libs/utils/decorate.js

```diff
@@ -260,3 +262,25 @@ export function applyInViewPortPlay(video) {
     observer.observe(video);
   }
 }
+
+export function turnAnchorIntoVideo({ hash, src, anchorTag }) {
+  const { dataset, parentElement } = anchorTag;
+  const attrs = getVideoAttrs(hash, dataset);
+  const video = `<video ${attrs}></video>`;
+  anchorTag.insertAdjacentHTML('afterend', video);
+  createIntersectionObserver({
+    el: parentElement,
+    options: { rootMargin: `${ROOT_MARGIN}px` },
+    callback: () => {
+      parentElement
+        .querySelector('video')
+        .appendChild(
+          createTag('source', { src, type: 'video/mp4' }),
+        );
+    },
+  });
+  const videoEl = parentElement.querySelector('video');
```

Comment:
Could we move this above the intersection observer and use `videoEl` on L275 instead of querying twice?

Reply:
Good idea, done 👍